batcher/aligned-batcher/src/connection.rs

```diff
@@ -66,3 +58,36 @@ pub(crate) async fn send_message<T: Serialize>(ws_conn_sink: WsMessageSink, mess
         Err(e) => error!("Error while serializing message: {}", e),
     }
 }
+
+pub(crate) async fn send_response(
+    ws_sink: &Arc<RwLock<SplitSink<WebSocketStream<TcpStream>, Message>>>,
+    serialized_response: Vec<u8>,
+) {
+    if let Err(e) = retry_function(
+        || send_response_retryable(ws_sink, serialized_response.clone()),
+        DEFAULT_MIN_DELAY,
+        DEFAULT_FACTOR,
+        DEFAULT_MAX_TIMES,
+    )
+    .await
+    {
+        error!("Error while sending batch inclusion data response: {}", e);
+    }
+}
+
+pub async fn send_response_retryable(
+    ws_sink: &Arc<RwLock<SplitSink<WebSocketStream<TcpStream>, Message>>>,
+    serialized_response: Vec<u8>,
+) -> Result<(), RetryError<Error>> {
+    let sending_result = ws_sink
+        .write()
+        .await
+        .send(Message::binary(serialized_response))
+        .await;
+
+    match sending_result {
+        Err(Error::AlreadyClosed) => Err(RetryError::Permanent(Error::AlreadyClosed)),
```

Comment:
Or maybe don't if we have a low `MaxRetry`

Reply:
good catch @MarcosNicolau, the `AlreadyClosed` error shouldn't be considered an error. It will raise in normal situations, so it should just be ignored:
```rust
Err(Error::AlreadyClosed) => ()
```
In other words, it should be an `Ok()` case.
The same applies for the `ConnectionClosed` error.

I think all other cases should be Permanent for the moment. You can see more details in https://docs.rs/tungstenite/latest/tungstenite/error/enum.Error.html

Reply:
Since all errors should be Permanent, I removed the retry logic from the `send_response()` method.

Reply:
Perfect!